GCC suggest-override warnings

Complete list of warnings when Boost 1.72 is built with GCC 7.5 with `-Wsuggest-override` added to `cxxflags`. Duplicated warnings from same location are omitted:

<pre>
./boost/wave/cpp_exceptions.hpp:164:25: warning: ‘virtual const char* boost::wave::preprocess_exception::what() const’ can be marked override [-Wsuggest-override]
./boost/wave/cpp_exceptions.hpp:168:25: warning: ‘virtual const char* boost::wave::preprocess_exception::description() const’ can be marked override [-Wsuggest-override]
./boost/wave/cpp_exceptions.hpp:172:17: warning: ‘virtual int boost::wave::preprocess_exception::get_severity() const’ can be marked override [-Wsuggest-override]
./boost/wave/cpp_exceptions.hpp:176:17: warning: ‘virtual int boost::wave::preprocess_exception::get_errorcode() const’ can be marked override [-Wsuggest-override]
./boost/wave/cpp_exceptions.hpp:180:25: warning: ‘virtual const char* boost::wave::preprocess_exception::get_related_name() const’ can be marked override [-Wsuggest-override]
./boost/wave/cpp_exceptions.hpp:184:18: warning: ‘virtual bool boost::wave::preprocess_exception::is_recoverable() const’ can be marked override [-Wsuggest-override]
./boost/wave/cpp_exceptions.hpp:384:25: warning: ‘virtual const char* boost::wave::macro_handling_exception::what() const’ can be marked override [-Wsuggest-override]
./boost/wave/cpp_exceptions.hpp:388:17: warning: ‘virtual const char* boost::wave::macro_handling_exception::get_related_name() const’ can be marked override [-Wsuggest-override]
./boost/wave/cpp_exceptions.hpp:79:25: warning: ‘virtual const char* boost::wave::cpp_exception::what() const’ can be marked override [-Wsuggest-override]
./boost/wave/cpplexer/cpplexer_exceptions.hpp:145:25: warning: ‘virtual const char* boost::wave::cpplexer::cpplexer_exception::what() const’ can be marked override [-Wsuggest-override]
./boost/wave/cpplexer/cpplexer_exceptions.hpp:189:25: warning: ‘virtual const char* boost::wave::cpplexer::lexing_exception::what() const’ can be marked override [-Wsuggest-override]
./boost/wave/cpplexer/cpplexer_exceptions.hpp:193:25: warning: ‘virtual const char* boost::wave::cpplexer::lexing_exception::description() const’ can be marked override [-Wsuggest-override]
./boost/wave/cpplexer/cpplexer_exceptions.hpp:197:17: warning: ‘virtual int boost::wave::cpplexer::lexing_exception::get_severity() const’ can be marked override [-Wsuggest-override]
./boost/wave/cpplexer/cpplexer_exceptions.hpp:201:17: warning: ‘virtual int boost::wave::cpplexer::lexing_exception::get_errorcode() const’ can be marked override [-Wsuggest-override]
./boost/wave/cpplexer/cpplexer_exceptions.hpp:205:18: warning: ‘virtual bool boost::wave::cpplexer::lexing_exception::is_recoverable() const’ can be marked override [-Wsuggest-override]
./boost/wave/cpplexer/re2clex/cpp_re2c_lexer.hpp:346:17: warning: ‘boost::wave::cpplexer::re2clex::lex_functor<IteratorT, PositionT, TokenT>::token_type& boost::wave::cpplexer::re2clex::lex_functor<IteratorT, PositionT, TokenT>::get(boost::wave::cpplexer::re2clex::lex_functor<IteratorT, PositionT, TokenT>::token_type&) [with IteratorT = char*; PositionT = boost::wave::util::file_position<boost::wave::util::flex_string<char, std::char_traits<char>, std::allocator<char>, boost::wave::util::CowString<boost::wave::util::AllocatorStringStorage<char> > > >; TokenT = boost::wave::cpplexer::lex_token<boost::wave::util::file_position<boost::wave::util::flex_string<char, std::char_traits<char>, std::allocator<char>, boost::wave::util::CowString<boost::wave::util::AllocatorStringStorage<char> > > > >; boost::wave::cpplexer::re2clex::lex_functor<IteratorT, PositionT, TokenT>::token_type = boost::wave::cpplexer::lex_token<boost::wave::util::file_position<boost::wave::util::flex_string<char, std::char_traits<char>, std::allocator<char>, boost::wave::util::CowString<boost::wave::util::AllocatorStringStorage<char> > > > >]’ can be marked override [-Wsuggest-override]
./boost/wave/cpplexer/re2clex/cpp_re2c_lexer.hpp:347:10: warning: ‘void boost::wave::cpplexer::re2clex::lex_functor<IteratorT, PositionT, TokenT>::set_position(const PositionT&) [with IteratorT = char*; PositionT = boost::wave::util::file_position<boost::wave::util::flex_string<char, std::char_traits<char>, std::allocator<char>, boost::wave::util::CowString<boost::wave::util::AllocatorStringStorage<char> > > >; TokenT = boost::wave::cpplexer::lex_token<boost::wave::util::file_position<boost::wave::util::flex_string<char, std::char_traits<char>, std::allocator<char>, boost::wave::util::CowString<boost::wave::util::AllocatorStringStorage<char> > > > >]’ can be marked override [-Wsuggest-override]
./boost/wave/cpplexer/re2clex/cpp_re2c_lexer.hpp:349:10: warning: ‘bool boost::wave::cpplexer::re2clex::lex_functor<IteratorT, PositionT, TokenT>::has_include_guards(std::__cxx11::string&) const [with IteratorT = char*; PositionT = boost::wave::util::file_position<boost::wave::util::flex_string<char, std::char_traits<char>, std::allocator<char>, boost::wave::util::CowString<boost::wave::util::AllocatorStringStorage<char> > > >; TokenT = boost::wave::cpplexer::lex_token<boost::wave::util::file_position<boost::wave::util::flex_string<char, std::char_traits<char>, std::allocator<char>, boost::wave::util::CowString<boost::wave::util::AllocatorStringStorage<char> > > > >; std::__cxx11::string = std::__cxx11::basic_string<char>]’ can be marked override [-Wsuggest-override]
</pre>

Depends on boostorg/config#329.

--- a/include/boost/wave/cpp_exceptions.hpp
+++ b/include/boost/wave/cpp_exceptions.hpp
@@ -76,7 +76,7 @@ public:
     }
     ~cpp_exception() throw() {}
 
-    virtual char const *what() const throw() = 0;           // to be overloaded
+    char const *what() const throw() BOOST_OVERRIDE = 0;    // to be overloaded
     virtual char const *description() const throw() = 0;
     virtual int get_errorcode() const throw() = 0;
     virtual int get_severity() const throw() = 0;
@@ -164,27 +164,27 @@ public:
     }
     ~preprocess_exception() throw() {}
 
-    virtual char const *what() const throw()
+    char const *what() const throw() BOOST_OVERRIDE
     {
         return "boost::wave::preprocess_exception";
     }
-    virtual char const *description() const throw()
+    char const *description() const throw() BOOST_OVERRIDE
     {
         return buffer;
     }
-    virtual int get_severity() const throw()
+    int get_severity() const throw() BOOST_OVERRIDE
     {
         return severity_level(code);
     }
-    virtual int get_errorcode() const throw()
+    int get_errorcode() const throw() BOOST_OVERRIDE
     {
         return code;
     }
-    virtual char const* get_related_name() const throw()
+    char const* get_related_name() const throw() BOOST_OVERRIDE
     {
         return "<unknown>";
     }
-    virtual bool is_recoverable() const throw()
+    bool is_recoverable() const throw() BOOST_OVERRIDE
     {
         switch (get_errorcode()) {
         // these are the exceptions thrown during processing not supposed to
@@ -395,11 +395,11 @@ public:
     }
     ~macro_handling_exception() throw() {}
 
-    virtual char const *what() const throw()
+    char const *what() const throw() BOOST_OVERRIDE
     {
         return "boost::wave::macro_handling_exception";
     }
-    char const* get_related_name() const throw()
+    char const* get_related_name() const throw() BOOST_OVERRIDE
     {
         return name;
     }

--- a/include/boost/wave/cpplexer/cpplexer_exceptions.hpp
+++ b/include/boost/wave/cpplexer/cpplexer_exceptions.hpp
@@ -142,7 +142,7 @@ public:
     }
     ~cpplexer_exception() throw() {}
 
-    virtual char const *what() const throw() = 0;   // to be overloaded
+    char const *what() const throw() BOOST_OVERRIDE = 0;   // to be overloaded
     virtual char const *description() const throw() = 0;
     virtual int get_errorcode() const throw() = 0;
     virtual int get_severity() const throw() = 0;
@@ -186,23 +186,23 @@ public:
     }
     ~lexing_exception() throw() {}
 
-    virtual char const *what() const throw()
+    char const *what() const throw() BOOST_OVERRIDE
     {
         return "boost::wave::lexing_exception";
     }
-    virtual char const *description() const throw()
+    char const *description() const throw() BOOST_OVERRIDE
     {
         return buffer;
     }
-    virtual int get_severity() const throw()
+    int get_severity() const throw() BOOST_OVERRIDE
     {
         return level;
     }
-    virtual int get_errorcode() const throw()
+    int get_errorcode() const throw() BOOST_OVERRIDE
     {
         return code;
     }
-    virtual bool is_recoverable() const throw()
+    bool is_recoverable() const throw() BOOST_OVERRIDE
     {
         switch (get_errorcode()) {
         case lexing_exception::universal_char_invalid:

--- a/include/boost/wave/cpplexer/re2clex/cpp_re2c_lexer.hpp
+++ b/include/boost/wave/cpplexer/re2clex/cpp_re2c_lexer.hpp
@@ -343,10 +343,10 @@ public:
     virtual ~lex_functor() {}
 
 // get the next token from the input stream
-    token_type& get(token_type& result) { return re2c_lexer.get(result); }
-    void set_position(PositionT const &pos) { re2c_lexer.set_position(pos); }
+    token_type& get(token_type& result) BOOST_OVERRIDE { return re2c_lexer.get(result); }
+    void set_position(PositionT const &pos) BOOST_OVERRIDE { re2c_lexer.set_position(pos); }
 #if BOOST_WAVE_SUPPORT_PRAGMA_ONCE != 0
-    bool has_include_guards(std::string& guard_name) const
+    bool has_include_guards(std::string& guard_name) const BOOST_OVERRIDE
         { return re2c_lexer.has_include_guards(guard_name); }
 #endif
 

--- a/samples/cpp_tokens/slex/cpp_slex_lexer.hpp
+++ b/samples/cpp_tokens/slex/cpp_slex_lexer.hpp
@@ -624,7 +624,7 @@ public:
     virtual ~slex_functor() {}
 
 // get the next token from the input stream
-    token_type& get(token_type& result)
+    token_type& get(token_type& result) BOOST_OVERRIDE
     {
         if (!at_eof) {
             do {
@@ -740,7 +740,7 @@ public:
         return result = token_type();   // return T_EOI
     }
 
-    void set_position(PositionT const &pos)
+    void set_position(PositionT const &pos) BOOST_OVERRIDE
     {
         // set position has to change the file name and line number only
         first.get_position().set_file(pos.get_file());
@@ -748,7 +748,7 @@ public:
     }
 
 #if BOOST_WAVE_SUPPORT_PRAGMA_ONCE != 0
-    bool has_include_guards(std::string& guard_name) const
+    bool has_include_guards(std::string& guard_name) const BOOST_OVERRIDE
         { return guards.detected(guard_name); }
 #endif
 

--- a/samples/cpp_tokens/slex/lexer.hpp
+++ b/samples/cpp_tokens/slex/lexer.hpp
@@ -117,16 +117,16 @@ public:
     char_node(const char_node& x);
     virtual ~char_node(){}
 
-    virtual node* clone() const;
-    virtual bool nullable() const;
-    virtual node_set firstpos() const;
-    virtual node_set lastpos() const;
-    virtual void compute_followpos(followpos_t& followpos) const;
-    virtual void compute_state_match(state_match_t& state_match ) const;
-    virtual void get_eof_ids(node_set& eof_set) const;
-    virtual void assign_node_ids(node_id_t& node_count);
+    node* clone() const BOOST_OVERRIDE;
+    bool nullable() const BOOST_OVERRIDE;
+    node_set firstpos() const BOOST_OVERRIDE;
+    node_set lastpos() const BOOST_OVERRIDE;
+    void compute_followpos(followpos_t& followpos) const BOOST_OVERRIDE;
+    void compute_state_match(state_match_t& state_match ) const BOOST_OVERRIDE;
+    void get_eof_ids(node_set& eof_set) const BOOST_OVERRIDE;
+    void assign_node_ids(node_id_t& node_count) BOOST_OVERRIDE;
 #if defined(BOOST_SPIRIT_DEBUG) && (BOOST_SPIRIT_DEBUG_FLAGS & BOOST_SPIRIT_DEBUG_FLAGS_SLEX)
-    virtual void dump(std::ostream& out) const;
+    void dump(std::ostream& out) const BOOST_OVERRIDE;
 #endif
 
 private:
@@ -233,16 +233,16 @@ public:
     epsilon_node(const epsilon_node& x);
     virtual ~epsilon_node(){}
 
-    virtual node* clone() const;
-    virtual bool nullable() const;
-    virtual node_set firstpos() const;
-    virtual node_set lastpos() const;
-    virtual void compute_followpos(followpos_t& followpos) const;
-    virtual void compute_state_match(state_match_t& state_match ) const;
-    virtual void get_eof_ids(node_set& eof_set) const;
-    virtual void assign_node_ids(node_id_t& node_count);
+    node* clone() const BOOST_OVERRIDE;
+    bool nullable() const BOOST_OVERRIDE;
+    node_set firstpos() const BOOST_OVERRIDE;
+    node_set lastpos() const BOOST_OVERRIDE;
+    void compute_followpos(followpos_t& followpos) const BOOST_OVERRIDE;
+    void compute_state_match(state_match_t& state_match ) const BOOST_OVERRIDE;
+    void get_eof_ids(node_set& eof_set) const BOOST_OVERRIDE;
+    void assign_node_ids(node_id_t& node_count) BOOST_OVERRIDE;
 #if defined(BOOST_SPIRIT_DEBUG) && (BOOST_SPIRIT_DEBUG_FLAGS & BOOST_SPIRIT_DEBUG_FLAGS_SLEX)
-    virtual void dump(std::ostream& out) const;
+    void dump(std::ostream& out) const BOOST_OVERRIDE;
 #endif
 
 private:
@@ -343,16 +343,16 @@ public:
     or_node(const or_node& x);
     virtual ~or_node(){}
 
-    virtual node* clone() const;
-    virtual bool nullable() const;
-    virtual node_set firstpos() const;
-    virtual node_set lastpos() const;
-    virtual void compute_followpos(followpos_t& followpos) const;
-    virtual void compute_state_match(state_match_t& state_match ) const;
-    virtual void get_eof_ids(node_set& eof_set) const;
-    virtual void assign_node_ids(node_id_t& node_count);
+    node* clone() const BOOST_OVERRIDE;
+    bool nullable() const BOOST_OVERRIDE;
+    node_set firstpos() const BOOST_OVERRIDE;
+    node_set lastpos() const BOOST_OVERRIDE;
+    void compute_followpos(followpos_t& followpos) const BOOST_OVERRIDE;
+    void compute_state_match(state_match_t& state_match ) const BOOST_OVERRIDE;
+    void get_eof_ids(node_set& eof_set) const BOOST_OVERRIDE;
+    void assign_node_ids(node_id_t& node_count) BOOST_OVERRIDE;
 #if defined(BOOST_SPIRIT_DEBUG) && (BOOST_SPIRIT_DEBUG_FLAGS & BOOST_SPIRIT_DEBUG_FLAGS_SLEX)
-    virtual void dump(std::ostream& out) const;
+    void dump(std::ostream& out) const BOOST_OVERRIDE;
 #endif
 
 private:
@@ -476,16 +476,16 @@ public:
     cat_node(const cat_node& x);
     virtual ~cat_node(){}
 
-    virtual node* clone() const;
-    virtual bool nullable() const;
-    virtual node_set firstpos() const;
-    virtual node_set lastpos() const;
-    virtual void compute_followpos(followpos_t& followpos) const;
-    virtual void compute_state_match(state_match_t& state_match ) const;
-    virtual void get_eof_ids(node_set& eof_set) const;
-    virtual void assign_node_ids(node_id_t& node_count);
+    node* clone() const BOOST_OVERRIDE;
+    bool nullable() const BOOST_OVERRIDE;
+    node_set firstpos() const BOOST_OVERRIDE;
+    node_set lastpos() const BOOST_OVERRIDE;
+    void compute_followpos(followpos_t& followpos) const BOOST_OVERRIDE;
+    void compute_state_match(state_match_t& state_match ) const BOOST_OVERRIDE;
+    void get_eof_ids(node_set& eof_set) const BOOST_OVERRIDE;
+    void assign_node_ids(node_id_t& node_count) BOOST_OVERRIDE;
 #if defined(BOOST_SPIRIT_DEBUG) && (BOOST_SPIRIT_DEBUG_FLAGS & BOOST_SPIRIT_DEBUG_FLAGS_SLEX)
-    virtual void dump(std::ostream& out) const;
+    void dump(std::ostream& out) const BOOST_OVERRIDE;
 #endif
 
 private:
@@ -632,16 +632,16 @@ public:
     star_node(const star_node& x);
     virtual ~star_node(){}
 
-    virtual node* clone() const;
-    virtual bool nullable() const;
-    virtual node_set firstpos() const;
-    virtual node_set lastpos() const;
-    virtual void compute_followpos(followpos_t& followpos) const;
-    virtual void compute_state_match(state_match_t& state_match ) const;
-    virtual void get_eof_ids(node_set& eof_set) const;
-    virtual void assign_node_ids(node_id_t& node_count);
+    node* clone() const BOOST_OVERRIDE;
+    bool nullable() const BOOST_OVERRIDE;
+    node_set firstpos() const BOOST_OVERRIDE;
+    node_set lastpos() const BOOST_OVERRIDE;
+    void compute_followpos(followpos_t& followpos) const BOOST_OVERRIDE;
+    void compute_state_match(state_match_t& state_match ) const BOOST_OVERRIDE;
+    void get_eof_ids(node_set& eof_set) const BOOST_OVERRIDE;
+    void assign_node_ids(node_id_t& node_count) BOOST_OVERRIDE;
 #if defined(BOOST_SPIRIT_DEBUG) && (BOOST_SPIRIT_DEBUG_FLAGS & BOOST_SPIRIT_DEBUG_FLAGS_SLEX)
-    virtual void dump(std::ostream& out) const;
+    void dump(std::ostream& out) const BOOST_OVERRIDE;
 #endif
 
 private:
@@ -755,16 +755,16 @@ public:
     eof_node(const eof_node& x);
     virtual ~eof_node(){}
 
-    virtual node* clone() const;
-    virtual bool nullable() const;
-    virtual node_set firstpos() const;
-    virtual node_set lastpos() const;
-    virtual void compute_followpos(followpos_t& followpos) const;
-    virtual void compute_state_match(state_match_t& state_match ) const;
-    virtual void get_eof_ids(node_set& eof_set) const;
-    virtual void assign_node_ids(node_id_t& node_count);
+    node* clone() const BOOST_OVERRIDE;
+    bool nullable() const BOOST_OVERRIDE;
+    node_set firstpos() const BOOST_OVERRIDE;
+    node_set lastpos() const BOOST_OVERRIDE;
+    void compute_followpos(followpos_t& followpos) const BOOST_OVERRIDE;
+    void compute_state_match(state_match_t& state_match ) const BOOST_OVERRIDE;
+    void get_eof_ids(node_set& eof_set) const BOOST_OVERRIDE;
+    void assign_node_ids(node_id_t& node_count) BOOST_OVERRIDE;
 #if defined(BOOST_SPIRIT_DEBUG) && (BOOST_SPIRIT_DEBUG_FLAGS & BOOST_SPIRIT_DEBUG_FLAGS_SLEX)
-    virtual void dump(std::ostream& out) const;
+    void dump(std::ostream& out) const BOOST_OVERRIDE;
 #endif
 
 private:
@@ -869,16 +869,16 @@ public:
     ccl_node(const ccl_node& x);
     virtual ~ccl_node(){}
 
-    virtual node* clone() const;
-    virtual bool nullable() const;
-    virtual node_set firstpos() const;
-    virtual node_set lastpos() const;
-    virtual void compute_followpos(followpos_t& followpos) const;
-    virtual void compute_state_match(state_match_t& state_match ) const;
-    virtual void get_eof_ids(node_set& eof_set) const;
-    virtual void assign_node_ids(node_id_t& node_count);
+    node* clone() const BOOST_OVERRIDE;
+    bool nullable() const BOOST_OVERRIDE;
+    node_set firstpos() const BOOST_OVERRIDE;
+    node_set lastpos() const BOOST_OVERRIDE;
+    void compute_followpos(followpos_t& followpos) const BOOST_OVERRIDE;
+    void compute_state_match(state_match_t& state_match ) const BOOST_OVERRIDE;
+    void get_eof_ids(node_set& eof_set) const BOOST_OVERRIDE;
+    void assign_node_ids(node_id_t& node_count) BOOST_OVERRIDE;
 #if defined(BOOST_SPIRIT_DEBUG) && (BOOST_SPIRIT_DEBUG_FLAGS & BOOST_SPIRIT_DEBUG_FLAGS_SLEX)
-    virtual void dump(std::ostream& out) const;
+    void dump(std::ostream& out) const BOOST_OVERRIDE;
 #endif
 
 private:

--- a/samples/list_includes/lexertl/lexertl_lexer.hpp
+++ b/samples/list_includes/lexertl/lexertl_lexer.hpp
@@ -608,7 +608,7 @@ public:
     ~lexertl_functor() {}
 
 // get the next token from the input stream
-    token_type& get(token_type& result)
+    token_type& get(token_type& result) BOOST_OVERRIDE
     {
         if (lexer_.is_initialized() && !at_eof) {
             do {
@@ -728,7 +728,7 @@ public:
         return result = token_type();           // return T_EOI
     }
 
-    void set_position(Position const &pos)
+    void set_position(Position const &pos) BOOST_OVERRIDE
     {
         // set position has to change the file name and line number only
         first.get_position().set_file(pos.get_file());
@@ -736,7 +736,7 @@ public:
     }
 
 #if BOOST_WAVE_SUPPORT_PRAGMA_ONCE != 0
-    bool has_include_guards(std::string& guard_name) const
+    bool has_include_guards(std::string& guard_name) const BOOST_OVERRIDE
         { return guards.detected(guard_name); }
 #endif
 

--- a/samples/token_statistics/xlex/xlex_lexer.hpp
+++ b/samples/token_statistics/xlex/xlex_lexer.hpp
@@ -526,11 +526,11 @@ public:
     virtual ~xlex_functor() {}
     
 // get the next token from the input stream
-    token_type& get(token_type& t) { return lexer_.get(t); }
-    void set_position(Position const &pos) { lexer_.set_position(pos); }
+    token_type& get(token_type& t) BOOST_OVERRIDE { return lexer_.get(t); }
+    void set_position(Position const &pos) BOOST_OVERRIDE { lexer_.set_position(pos); }
 
 #if BOOST_WAVE_SUPPORT_PRAGMA_ONCE != 0
-    bool has_include_guards(std::string& guard_name) const 
+    bool has_include_guards(std::string& guard_name) const BOOST_OVERRIDE
         { return lexer_.has_include_guards(guard_name); }
 #endif    
 

--- a/tool/trace_macro_expansion.hpp
+++ b/tool/trace_macro_expansion.hpp
@@ -85,15 +85,15 @@ public:
     }
     ~bad_pragma_exception() throw() {}
 
-    virtual char const *what() const throw()
+    char const *what() const throw() BOOST_OVERRIDE
     {
         return "boost::wave::bad_pragma_exception";
     }
-    virtual bool is_recoverable() const throw()
+    virtual bool is_recoverable() const throw() BOOST_OVERRIDE
     {
         return true;
     }
-    virtual int get_severity() const throw()
+    virtual int get_severity() const throw() BOOST_OVERRIDE
     {
         return boost::wave::util::severity_remark;
     }

